src/View/Helper/FormHelper.php

```diff
@@ -191,6 +191,14 @@ public function input($fieldName, array $options = [])
         $options = $this->_parseOptions($fieldName, $options);
         $reset = $this->templates();
 
+        $templater = $this->templater();
+        if (!empty($options['templates'])) {
+            $templater->push();
+            $templateMethod = is_string($options['templates']) ? 'load' : 'add';
+            $templater->{$templateMethod}($options['templates']);
+        }
+        unset($options['templates']);
```

Comment:
```php
public function input($fieldName, array $options = [])
    {
        $options += [
            'prepend' => null,
            'append' => null,
            'type' => null,
            'label' => null,
            'error' => null,
            'required' => null,
            'options' => null,
            'help' => null,
            'templates' => [],
            'templateVars' => []
        ];
        $options = $this->_parseOptions($fieldName, $options);
        $reset = $this->templates();

        $templater = $this->templater();
        if (!empty($options['templates'])) {
            $templater->push();
            $templateMethod = is_string($options['templates']) ? 'load' : 'add';
            $templater->{$templateMethod}($options['templates']);
        }
        unset($options['templates']);
```
The function always throws in an empty templates array even when the passed $options doesn't have it.  I was checking to see if it is !empty() because it's always going to be set.  This chunk of code comes from Cake's original Form Helper.  

The only thing I'm unsure of now is whether or not the ```unset($options['templates']);``` should be there or not.  Not sure if there would be a need to have those templates passed on through to Cake's Form Helper.